internal/resolver/unix/unix.go

```diff
@@ -37,7 +37,20 @@ func (b *builder) Build(target resolver.Target, cc resolver.ClientConn, _ resolv
 	if target.Authority != "" {
 		return nil, fmt.Errorf("invalid (non-empty) authority: %v", target.Authority)
 	}
-	addr := resolver.Address{Addr: target.Endpoint}
+
+	// gRPC was parsing the dial target manually before PR #4817, and we
+	// switched to using url.Parse() in that PR. To avoid breaking existing
+	// resolver implementations we ended up stripping the leading "/" from the
+	// endpoint. This obviously does not work for the "unix" scheme. Hence we
+	// end up using the parsed URL instead.
+	endpoint := target.Endpoint
+	if u := target.ParsedURL; u != nil {
```

Comment:
Why would `ParsedURL` be `nil`?  Don't we guarantee it's always set?

I think the `nil` check is fine as defensive programming to avoid a panic, but don't fall back to `Endpoint` I'd say; just return an error.

```go
if target.ParsedURL == nil {
	return fmt.Errorf(...)
}
```

Or maybe include `ParsedURL` by value instead?  That might be the cleanest way to go.

Reply:
I switched the `nil` check to return an error as per your suggestion. Yes, we do guarantee that it is always set.

I'm not super comfortable storing the `url.URL` by value in `resolver.Target` since we don't own that type and all methods on the type have a pointer receiver.

Just curious. Why do you think that storing it by value might be the cleanest way to go?

Reply:
> I'm not super comfortable storing the url.URL by value in resolver.Target since we don't own that type

This would matter if there is a mutex in there that makes it un-copyable, but since there is none now, then it would be backward-incompatible for the Go library to change that in the future.  I think this is safe.

> and all methods on the type have a pointer receiver.

These methods still work fine when called on a value.  Almost all receivers are on pointers instead of values to avoid unnecessary copying when calling.

> Why do you think that storing it by value might be the cleanest way to go?

Because then there is no need to check for `nil`-ness of something we guarantee will not be `nil`.  That's one reason why we pass/return all our options/etc structs in the balancer/resolver packages as values instead of pointers.  (The other big reason is that they are "immutable" (by the other party).  This isn't something that concerns me much, though it is troubling to think what might happen if the resolver did `target.ParsedURL.Host = "nope"` since it can technically modify the CC's copy of the `ParsedURL` this way.)

Reply:
Thanks. Done.